code/modules/projectiles/projectile/magic.dm

```diff
@@ -161,7 +161,14 @@
 		M = shapeshift.stored
 		shapeshift.restore()
 
-	if(!istype(M) || M.stat == DEAD || M.notransform || (GODMODE & M.status_flags))
+	if(!istype(M) || istype(M, /mob/living/simple_animal/hostile/mimic/copy) || M.stat == DEAD || M.notransform || (GODMODE & M.status_flags))
+		return
+
+	if(ispAI(M))
```

Comment:
Instead of type checking for pAI or mimic, could you move this to a redefined proc?
Something like `do_wabbajack()` that is redefined to do nothing for the mimic, and to change the chasis for the pai.

Reply:
Done with 7a08511a40f0d220f60c8cfc3e38d2483e599cee